### PR TITLE
Simplifies and update versions

### PR DESCRIPTION
**Pull request by codefromthecrypt**
This updates to..

* a simpler Dockerfile including more recent versions
* use latest docker-plugins to apply new release policy https://github.com/hypertrace/hypertrace/issues/19
* use Docker 19 by default in CircleCI https://github.com/hypertrace/hypertrace/issues/22
* use a more recent Java image when executing Docker via Grade

### CODE REVIEW

**Comment by aaron-steinfeld** [comment]
Not sure I follow the motivation for these two changes - is there something in this dockerfile (or a desired change) that's not supported in 17? And why switch to java 14? Neither should cause an issue with this build, but makes this build unique for no reason I'm aware of.

Also, looks like supported version would be `19.03.12` but not `19.03.11`:  https://circleci.com/docs/2.0/building-docker-images/#docker-version

**Comment by codefromthecrypt** [comment]
@aaron-steinfeld I raised this too early as I wanted to adjust for this first https://github.com/hypertrace/kafka/pull/9

It is unnecessary for so many different projects to have different plugin configuration, or setup traps that allow things like the USER bug to happen by default in all repos.  Once the above is in, we should update all our projects vs wait for someone to fall into each trap independently.

**Comment by aaron-steinfeld** [comment]
> It is unnecessary for so many different projects to have different plugin configuration, or setup traps that allow things like the USER bug to happen by default in all repos. Once the above is in, we should update all our projects vs wait for someone to fall into each trap independently.

Agree, migrating all the projects is a good idea and sufficient motivation here. Thought it was being singled out for some reason.

**Comment by codefromthecrypt** [comment]
yep I made a mistake as created the PR before the meta-issue. will not do that again as there was no value from me doing it backwards https://github.com/hypertrace/hypertrace/issues/22

**Comment by codefromthecrypt** [comment]
I've reworked this PR to combine all currently known outstanding efforts, and updated and polished the dockerfile while at it.

File: Dockerfile [hunk 1]
```diff
@@ -1,14 +1,16 @@
...
 
 # Install Helm
-RUN curl -Lo helm-v3.1.2-linux-amd64.tar.gz https://get.helm.sh/helm-v3.1.2-linux-amd64.tar.gz && \
-    tar -zxvf helm-v3.1.2-linux-amd64.tar.gz && \
-    mv linux-amd64/helm /usr/local/bin/helm && \
-    rm helm-v3.1.2-linux-amd64.tar.gz && \
-    rm -rf linux-amd64 && \
-    helm version
+RUN curl -sSL https://get.helm.sh/helm-v${HELM_VERSION}-linux-amd64.tar.gz | tar xz \
+  --strip=1 linux-amd64/helm && \
```
Comment (codefromthecrypt) [inline Dockerfile]: neat trick is strip removes the usual dance of extracting something into a dir then later removing the junk around it. This is especially easy when the tar includes a statically linked binary

**Review by aaron-steinfeld — DISMISSED** [review]

File: Dockerfile [hunk 2]
```diff
@@ -1,14 +1,16 @@
 FROM alpine:3.12
 
-RUN apk --no-cache add curl git
+RUN apk --update --no-cache add curl git
```
Comment (jcchavezs) [inline Dockerfile]: being very picky here is if this container's size is important that makes sense to uninstall curl and git afterwards.

Comment (codefromthecrypt) [reply Dockerfile]: that part I don't know about. @aaron-steinfeld do you know about this? if we don't need curl or git after I agree and will uninstall and also remove from the README

Comment (aaron-steinfeld) [reply Dockerfile]: I don't believe we need to keep curl, but since this is a build image that will need to check out code, including git makes sense

Comment (aaron-steinfeld) [reply Dockerfile]: @ravisingal do you remember if there was a specific reason for curl? I can't recall one but it looks like it was included explicitly. Unless we can recall a reason, I think it's fair to assume it was just added in order to run the later steps of the dockerfile.

Comment (ravisingal) [reply Dockerfile]: yes, it was added to run the next step.

**Comment by codefromthecrypt** [comment]
removed use of curl. thanks folks.

**Comment by codefromthecrypt** [comment]
oops looks like the plugin thing also uses curl. will do it differently

**Comment by codefromthecrypt** [comment]
ok it is a good bit smaller now:

```
$ docker images|grep hypertrace/helm-gcs-packager
hypertrace/helm-gcs-packager   test                cb0bcdc6f392        27 seconds ago      135MB
hypertrace/helm-gcs-packager   0.1.3               c3d0a429bf09        3 weeks ago         113MB
```

**Comment by codefromthecrypt** [comment]
lol whoops it isn't :P

**Comment by codefromthecrypt** [comment]
Looks like there are some reorganizations in how plugins work..

before:
```
$ du -sk $HOME/.cache $HOME/.local
51688	/root/.cache
20	/root/.local
$ ls -al /root/.local/share/helm/plugins/helm-gcs.git
lrwxrwxrwx    1 root     root            63 Jul 30 20:50 /root/.local/share/helm/plugins/helm-gcs.git -> /root/.cache/helm/plugins/https-github.com-hayorov-helm-gcs.git
```

after
```
$ du -sk $HOME/.cache $HOME/.local
20188	/root/.cache
51716	/root/.local
$ ls -al /root/.local/share/helm/plugins/helm-gcs.git
total 164
drwxr-xr-x    9 root     root          4096 Aug 25 03:55 .
drwxr-xr-x    3 root     root          4096 Aug 25 03:55 ..
drwxr-xr-x    8 root     root          4096 Aug 25 03:55 .git
drwxr-xr-x    2 root     root          4096 Aug 25 03:55 .github
-rw-r--r--    1 root     root           312 Aug 25 03:55 .gitignore
-rw-r--r--    1 root     root           556 Aug 25 03:55 .goreleaser.yml
-rw-r--r--    1 root     root          1083 Aug 25 03:55 .travis.yml
--snip--
```

**Comment by codefromthecrypt** [comment]
looks like helm plugin duplicates the git subtree

```
$ du -sk /root/.local/share/helm/plugins/helm-gcs.git/.git
19924	/root/.local/share/helm/plugins/helm-gcs.git/.git
$ du -sk /root/.cache/helm/plugins/https-github.com-hayorov-helm-gcs.git/.git
19924	/root/.cache/helm/plugins/https-github.com-hayorov-helm-gcs.git/.git
```

Anyone opposed to just nuking both of them :) I don't think helm uses this at runtime does it?

**Comment by codefromthecrypt** [comment]
ok cleaned up the cache and .git dir as we won't be live updating the image

```
$ docker images|grep helm-gcs-packager 
hypertrace/helm-gcs-packager   test                a46629a11fe0        11 seconds ago       94.2MB
hypertrace/helm-gcs-packager   0.1.3               c3d0a429bf09        3 weeks ago          113MB
```

**Review by ravisingal — APPROVED** [review]

**Review by aaron-steinfeld — APPROVED** [review]

**Comment by codefromthecrypt** [comment]
thanks for all the eyes here!

**Comment by aaron-steinfeld** [comment]
@adriancole FYI this didn't publish, but not due to the change in release policy as discussed in https://github.com/hypertrace/hypertrace/issues/19 , but due to a glitch caused by https://github.com/hypertrace/hypertrace/issues/18 . It appears a small number of repos are affected by:
https://discuss.circleci.com/t/circleci-doesnt-notice-when-github-default-branch-is-updated/10469 (not exactly sure the reason why some are affected and not others, my best guess is something to do with in flight PRs that were retargeted vs new PRs). Anyway I reset circle for this project and it should be able to publish now. I can trigger the publish via API, but it would follow the old versioning rules that I discussed in the linked issue. 

I'll put up a change that we can potentially use as a template for other repos and if that looks good, we can follow the new process here and release as 0.2.0. Else we can fallback and trigger via API (`POST https://circleci.com/api/v2/project/gh/hypertrace/helm-gcs-packager/pipeline` with a `Circle-Token` header using your api token)

**Comment by aaron-steinfeld** [comment]
https://github.com/hypertrace/helm-gcs-packager/pull/6

**Comment by codefromthecrypt** [comment]
thanks for chasing this up @aaron-steinfeld

**Pull request merged**